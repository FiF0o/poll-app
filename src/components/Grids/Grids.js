/**
 * Created by jonlazarini on 27/04/17.
 */
import React from 'react';
import { GridList, GridTile } from 'material-ui/GridList';
import Subheader from 'material-ui/Subheader';


const styles = {
    root: {
        display: 'flex',
        flexWrap: 'wrap',
        justifyContent: 'space-around',
    },
    gridList: {
        width: 480,
        height: 'auto',
        overflowY: 'auto',
    },
};

const tilesData = [
    {
        img: 'http://placehold.it/640x640',
        title: 'Breakfast',
        author: 'jill111',
        titlePosition: 'top'
    },
    {
        img: 'http://placehold.it/641x640',
        title: 'Tasty burger',
        author: 'pashminu',
        titlePosition: 'top'
    },
    {
        img: 'http://placehold.it/642x640',
        title: 'Tasty',
        author: 'painu',
        fullBleed: true,
        titlePosition: 'bottom'
    },
    {
        img: 'http://placehold.it/1280x640',
        title: 'Tasty',
        author: 'painu',
        fullBleed: true,
        titlePosition: 'bottom'
    },
];

/**
 * A simple example of a scrollable `GridList` containing a [Subheader](/#/components/subheader).
 */
const GridsExmple = () => (
    <div style={styles.root}>
        <GridList
            cellHeight={180}
            style={styles.gridList}
        >
            <Subheader>December</Subheader>
            {tilesData.map((tile) => (
                <GridTile
                    key={tile.img}
                    title={tile.title}
                    subtitle={<span>by <b>{tile.author}</b></span>}
                    cols={ tile.fullBleed ? 2 : 1 }
                    rows={ tile.fullBleed ? 2 : 1 }
                    titlePosition={ tile.titlePosition ? tile.titlePosition : 'bottom' }
                >
                    <img
                        src={tile.img}
                        alt={tile.img}
                    />
                </GridTile>
            ))}
        </GridList>
    </div>
);

export default GridsExmple;