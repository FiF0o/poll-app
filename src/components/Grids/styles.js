/**
 * Created by jonlazarini on 03/06/17.
 */
export const styles = {
    root: {
        display: 'flex',
        flexWrap: 'wrap',
        justifyContent: 'space-around',
    },
    gridList: {
        width: '100%',
        height: 'auto',
        // overflowY: 'auto',
    },
};
