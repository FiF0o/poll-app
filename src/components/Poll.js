/**
 * Created by jonlazarini on 24/04/17.
 */
import React, { Component } from 'react';

import { NewButton } from './Button';


class Poll extends Component {
    constructor(props) {
        super(props);

        this.state = {
            data: null,
            newData: null
        }
    }

    componentDidMount() {

    }

    render() {

        return(
            <div>
                <NewButton />
            </div>
        )
    }

}

export default Poll
