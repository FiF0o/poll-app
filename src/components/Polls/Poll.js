/**
 * Created by jonlazarini on 20/05/17.
 */
import React from 'react';
import {Card, CardActions, CardTitle, CardText} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';

import {UserAvatar} from '../Avatars';
import {Votes} from '../Votes';

// import {getVotesForPoll} from '../../utils/PollUtils';
import {isTheCurrentUser} from '../../utils/UserAuth';


export const Poll = ({pollKey, name, description, author, id, auth, pollUid, user, onDeletePoll, addVote, removeVote, polls, ...props}) => (
    <article>
        <UserAvatar user={user} />
        <Card>
            <CardTitle title={name} />
            <CardText>
                <b>Description:</b><br/>
                { description }<br/><br/>
            </CardText>
            <CardActions>
                {
                    isTheCurrentUser(pollUid, auth.uid) ? <FlatButton label="Delete My Poll" onTouchTap={onDeletePoll} /> : null
                }
                {/*votes={props.votes}*/}
                <Votes
                    voters={props.voters}
                    addVote={addVote}
                    removeVote={removeVote}
                    pollId={id}
                    auth={auth}
                    polls={polls}
                />
            </CardActions>

        </Card>
    </article>
);
