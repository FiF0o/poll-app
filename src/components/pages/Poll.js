/**
 * Created by jonlazarini on 24/04/17.
 */
import React, { Component } from 'react';
import CircularProgress from 'material-ui/CircularProgress';

import { database } from '../../database/firebase';

import Form from '../Form';
import { NewButton } from '../Button';

/**
 * "rules":{ ".read": "true" , ".write": "true"}
 */
try {
    // debug - gets /hello node
    database.ref('/').child('/hello').on('value', (snapshot) => {
        console.log('ON - hello node: ', snapshot.val());
    });
}
catch(err) {
    console.log(err)
}


class Poll extends Component {
    constructor(props) {
        super(props);

        this.state = {
            data: null,
            newData: null,
            hasLoaded: false,
            hasError: false,
            error: 'no error!'
        }

    }

    componentDidMount() {
        database.ref('/').once('value')
            .then((snapshot) => {
                /* debug error handling */
                // throw new Error('No bueno!');
                const data = snapshot.val();

                this.setState({
                    data: data,
                    hasLoaded: !this.state.hasLoaded
                });
            })
            .catch((err) => {
                console.error(err)

                this.setState({
                    hasError: !this.state.hasLoaded,
                    error: err.message
                });
            });
    }

    render() {

        // const { hello } = this.state.data
        // console.log(hello)

        if(this.state.hasError) {
            return (
                <div className="poll-page poll-page-error">{ this.state.error }</div>
            )
        }

        return (
            <section className="poll-page">
                <div className="poll-page poll-page-content">
                    {
                        !this.state.hasLoaded ?
                            <CircularProgress size={60} thickness={7}/>
                       :
                            <div className="">
                                <Form />
                                { JSON.stringify(this.state.data, null, 2) }
                                <NewButton />
                            </div>
                    }
                </div>
            </section>
        )
    }

}

export default Poll
