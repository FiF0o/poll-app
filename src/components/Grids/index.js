/**
 * Created by jonlazarini on 03/06/17.
 */
import React from 'react';
import { GridList } from 'material-ui';
import Subheader from 'material-ui/Subheader';
import {styles} from './styles';

export const Grid = ({title, cols, padding, cellHeight, children}) => (
    <div style={styles.root}>
        <Subheader><b>{title}</b></Subheader>
        <GridList
            style={styles.gridList}
            cellHeight={cellHeight}
            cols={cols}
            padding={padding}
        >
            {children}
        </GridList>
    </div>
);
