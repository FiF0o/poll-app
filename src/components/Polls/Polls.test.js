import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { map }from 'lodash';
import {Poll} from './Poll';
import {Grid} from '../Grids/';
import {Polls} from './'

describe('<Polls /> component', () => {
    
    it('should render correctly', () => {
        const pollsStub = {
            user: {
                polls: [{name: '', description: '', author: '', uid: '', id: ''}],
                auth: 'a@mail.com',
                users: [],
                deletePoll: null,
                addVote: null,
                removeVote: null,
            }
        }
        const tree = renderer.create(
            <MuiThemeProvider>
                <Polls {...pollsStub} />
            </MuiThemeProvider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})