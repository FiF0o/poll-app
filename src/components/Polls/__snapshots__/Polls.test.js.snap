// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Polls /> component should render correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
      "justifyContent": "space-around",
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.54)",
        "fontSize": 14,
        "fontWeight": 500,
        "lineHeight": "48px",
        "muiPrepared": true,
        "paddingLeft": 16,
        "width": "100%",
      }
    }
  >
    <b />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "height": "auto",
        "margin": -25,
        "muiPrepared": true,
        "width": "100%",
      }
    }
  />
</div>
`;
